diktat-rules/src/main/kotlin/com/saveourtool/diktat/ruleset/rules/chapter2/kdoc/KdocFormatting.kt

```diff
@@ -237,6 +237,19 @@ class KdocFormatting(configRules: List<RulesConfig>) : DiktatRule(
     private fun checkEmptyLineBeforeBasicTags(basicTags: List<KDocTag>) {
         val firstBasicTag = basicTags.firstOrNull() ?: return
 
+        // Unlike other tags, @property and @constructor tags are inside separate KDOC_SECTION,
+        // so we need to move all children from current KDOC_SECTION to previous one and delete current KDOC_SECTION
+        if (firstBasicTag.knownTag == KDocKnownTag.PROPERTY || firstBasicTag.knownTag == KDocKnownTag.CONSTRUCTOR) {
+            val currentSection = firstBasicTag.node.treeParent
+            val prevSection = currentSection.prevSibling { it.elementType == KDocElementTypes.KDOC_SECTION }
+
+            prevSection?.let {
+                prevSection.addChildren(currentSection.firstChildNode, currentSection.lastChildNode, null)
+                prevSection.addChild(currentSection.lastChildNode, null)
```

Comment:
wait, on the above statement, you had already added `currentSection.lastChildNode` to `prevSection`, no?

Reply:
The `addChildren` method is implemented in such way that last child is not added, so I needed to add it separately:
```kotlin
        ASTNode next;
        for(ASTNode f = firstChild; f != lastChild; f = next) {
            next = f.getTreeNext();
            this.addChild(f, anchorBefore);
        }
```